src/components/TodoList.tsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { MutableRefObject } from 'react';
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+
+type Props = {
+  tempTodo: Todo | undefined;
+  todoForDelete: number[];
+  setFocusForm: (isFocused: boolean) => void;
+  handleToggle: (todo: Todo | undefined) => void;
+  getFilteredTodos: Todo[];
+  renaming: Todo | undefined;
+  setNewTitle: (newTitle: string) => void;
+  newTitle: string;
+  loading: boolean;
+  handleRename: (event?: React.FormEvent<HTMLFormElement>) => void;
+  setRenaming: (todo: Todo | undefined) => void;
+  handleDelete: (id: number | undefined) => void;
+  formRef: MutableRefObject<HTMLInputElement | null>;
+};
+
+export const TodoList: React.FC<Props> = ({
+  tempTodo,
+  todoForDelete,
+  setFocusForm,
+  getFilteredTodos,
+  renaming,
+  handleToggle,
+  handleRename,
+  setNewTitle,
+  newTitle,
+  loading,
+  handleDelete,
+  formRef,
+  setRenaming,
```

Comment:
There are a lot of props which can be merged, like updating todo or props for showing loader. Consider to  fix it.